sdk/search/azure-search/src/test/java/com/azure/search/test/environment/setup/AzureSearchResources.java

```diff
@@ -113,9 +115,10 @@ public void deleteService() {
     /**
      * Creates the Resource Group in Azure. This should be run at @BeforeAll
      */
-    public void createResourceGroup(TestResourceNamer testResourceNamer) {
+    public void createResourceGroup() {
         if (resourceGroup == null) {
-            String resourceGroupName = testResourceNamer.randomName(RESOURCE_GROUP_NAME_PREFIX, 24);
+            String resourceGroupName = Configuration.getGlobalConfiguration().get(AZURE_RESOURCEGROUP_NAME);
+            //String resourceGroupName = testResourceNamer.randomName(RESOURCE_GROUP_NAME_PREFIX, 24);
```

Comment:
We may need to create a resource group name, will need to handle logic slightly differently here.

When tests are running in an Azure Pipeline we should trust that the resource group name is associated to an existing resource group. If there isn't a resource group environment variable we need to create the resource group, this will be needed when running tests locally. Additionally, running tests locally could support using the environment variable for resource group but it shouldn't create the group, if it doesn't exist it is on the person running the tests but we need to document this well.

Reply:
Change the logic here to 
if  resourceGroup exists, get the resource group.
If resourceGroup not exist, create an resource group.

By doing this, the entire test run only needs one resource group, so it does not need to delete every test class, which answered your question below.